File: qa-fundamentals/OrderManagementJava/src/main/java/org/orders/model/Stock.java
package org.orders.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "stock")
@Getter
@Setter
public class Stock {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "stock_id")
    private Integer stockId;

    @NotNull
    @Column(name = "product_id", nullable = false)
    private Integer productId;

    @NotNull
    @Min(value = 0, message = "Quantity must be at least 0")
    @Column(name = "quantity", nullable = false)
    private Integer quantity;

    @Column(name = "created_at")
    private java.time.LocalDateTime createdAt;

    @Column(name = "updated_at")
    private java.time.LocalDateTime updatedAt;

    // Navigation property
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", insertable = false, updatable = false)
    private Product product;

    // JPA lifecycle callbacks for timestamps
    @PrePersist
    protected void onCreate() {
        createdAt = java.time.LocalDateTime.now();
        updatedAt = java.time.LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = java.time.LocalDateTime.now();
    }
}

File: backend-intermediate-cs/OrderManagement/Models/Stock.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OrderManagement.Models {
  [Table("stock")]
  public class Stock {
    [Key]
    [Column("stock_id")]
    public int StockId { get; set; }
    

    [Column("product_id")]
    public int ProductId { get; set; }
    
    [Required]
    [Column("quantity")]
    [Range(0, int.MaxValue, ErrorMessage = "Quantity must be at least 0")]
    public int Quantity { get; set; }
    
    [Column("created_at")]
    public DateTime CreatedAt { get; set; }
    
    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; }
    
    // Navigation property
    public virtual Product? Product { get; set; }
  }
}
